client/network/src/protocol.rs

```diff
@@ -1081,11 +1084,15 @@ impl<B: BlockT, H: ExHashT> Protocol<B, H> {
 		self.transaction_pool.on_broadcasted(propagated_to);
 	}
 
+	pub fn announce_preimport_block(&mut self, header: B::Header) {
+		self.broadcast_header(header, false, None);
```

Comment:
If we don't announce this as best block, the other peers will not request this block. And no, the solution is not to pass `true` here.

You need to check the `BlockImportParams::ForkChoice` to check if the block is a new best block.

Reply:
The announcement for an imported block on the networking layer looks at:
```
let is_best = self.chain.info().best_hash == header.hash();
```
To figure out whether a block is best or not. At pre-import, how can `BlockImportParams::ForkChoice` be helpful if we haven't imported the block yet?